src/main/java/org/opensearch/neuralsearch/processor/rerank/ByFieldRerankProcessor.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.rerank;
+
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.common.collect.Tuple;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.common.bytes.BytesReference;
+import org.opensearch.core.xcontent.MediaType;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.neuralsearch.processor.rerank.context.ContextSourceFetcher;
+import org.opensearch.search.SearchHit;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+public class ByFieldRerankProcessor extends RescoringRerankProcessor {
+
+    public static final String TARGET_FIELD = "target_field";
+    public static final String REMOVE_TARGET_FIELD = "remove_target_field";
+
+    protected final String targetField;
+    protected final boolean removeTargetField;
+
+    /**
+     * Constructor. pass through to RerankProcessor constructor.
+     *
+     * @param description
+     * @param tag
+     * @param ignoreFailure
+     * @param targetField           the field you want to replace your score with
+     * @param removeTargetField
+     * @param contextSourceFetchers
+     */
+    public ByFieldRerankProcessor(
+        String description,
+        String tag,
+        boolean ignoreFailure,
+        String targetField,
+        boolean removeTargetField,
+        final List<ContextSourceFetcher> contextSourceFetchers
+    ) {
+        super(RerankType.BY_FIELD, description, tag, ignoreFailure, contextSourceFetchers);
+        this.targetField = targetField;
+        this.removeTargetField = removeTargetField;
+    }
+
+    @Override
+    public void rescoreSearchResponse(SearchResponse response, Map<String, Object> rerankingContext, ActionListener<List<Float>> listener) {
+        SearchHit[] searchHits = response.getHits().getHits();
+
+        if (!searchHitsHaveValidForm(searchHits, listener)) {
+            return;
+        }
+
+        List<Float> scores = new ArrayList<>(searchHits.length);
+
+        for (SearchHit hit : searchHits) {
+            Tuple<? extends MediaType, Map<String, Object>> typeAndSourceMap = getMapTuple(hit);
+            Map<String, Object> sourceAsMap = typeAndSourceMap.v2();
+
+            Object val = getValueFromMap(sourceAsMap, targetField).get();
+            scores.add(((Number) val).floatValue());
+
+            sourceAsMap.put("previous_score", hit.getScore());
+            if (removeTargetField) {
```

Comment:
Add comment on why this check is here. Also what does removetargetField do.

Reply:
Right, this is the feature of the ByField Processor (It essentially gives the user to delete a field they have used to update  the `_score` field), I have updated the constructor to better explain was in a rush to push the draft pr 😅